lib/mongo/server/connection_pool/generation_manager.rb

```diff
@@ -44,10 +46,28 @@ def generation_unlocked(service_id: nil)
           @map[service_id]
         end
 
+        def pipe_fds(service_id: nil)
+          @pipe_fds[service_id][@map[service_id]]
+        end
+
+        def remove_pipe_fds(generation, service_id: nil)
+          validate_service_id!(service_id)
+
+          r, w = @pipe_fds[service_id].delete(generation)
+          w.close
+          # Schedule the read end of the pipe to be closed. We cannot close it
+          # immediately since we need to wait for the Kernel#select call to
```

Comment:
1. I think we should clarify here where the `select` call is.
2. I am not sure I understand how we wait for the `select` call to return.